src/ansys/edb/core/definition/padstack_def_data.py

```diff
@@ -473,3 +503,47 @@ def solder_ball_material(self, material):
                 self, material
             )
         )
+
+    def get_connection_pt(self, layer):
+        """
+        Get connection point position and direction by layer name.
+
+        Parameters
+        ----------
+        layer : str
+            Layer name.
+
+        Returns
+        -------
+        tuple[:class:`geometry.PointData`, :class:`ConnectionPtDirection`]
+
+        The tuple is in a ``(position, direction)`` format:
+
+        - ``position``: Position of the connection point.
+        - ``direction``: Direction of the connection point.
+        """
+        msg = self.__stub.GetConnectionPt(
+            _PadstackDefDataQueryBuilder.padstack_def_data_get_connection_pt_message(self, layer)
+        )
+        return parser.to_point_data(msg), ConnectionPtDirection(msg.direction)
+
+    def set_connection_pt(self, layer, x, y, direction):
```

Comment:
maybe change the `x, y` arguments to `pt` to be consistent (see `conversions.to_point` for how to handle input of type PointLike)

Reply:
x, y arguments are replaced by position (PointData)
```
circ1: PadstackDef = PadstackDef.find_by_name(db, "Circ1")
circ1_data: PadstackDefData = circ1.data
circ1_p1 = circ1_data.get_connection_pt("Trace")

position1 = PointData((3,3))
direction1 = ConnectionPtDirection.PS_45_DIRECTION
circ1_data.set_connection_pt("Trace", position1, direction1)
```

Reply:
can you add `conversions.to_point` as well? for basic and geometry values (value, point, polygon etc) we generally try to avoid strong(er) typing and catch all possible input types.

```py
x = 1e-6
y = 1e-6
# covers raw types like
set_connection_pt(layer, (x, y), PS_45_DIRECTION)
# instead of just
set_connection_pt(layer, PointData(x, y), PS_45_DIRECTION)
```